src/Api/Controller/UploadFaviconController.php

```diff
@@ -10,68 +10,44 @@
 namespace Flarum\Api\Controller;
 
 use Flarum\Settings\SettingsRepositoryInterface;
-use Illuminate\Support\Arr;
-use Illuminate\Support\Str;
+use Intervention\Image\Image;
 use Intervention\Image\ImageManager;
 use League\Flysystem\FilesystemInterface;
-use Psr\Http\Message\ServerRequestInterface;
-use Tobscure\JsonApi\Document;
+use Psr\Http\Message\UploadedFileInterface;
 
-class UploadFaviconController extends ShowForumController
+class UploadFaviconController extends UploadImageController
 {
     /**
-     * @var SettingsRepositoryInterface
-     */
-    protected $settings;
-
-    /**
-     * @var FilesystemInterface
-     */
-    protected $uploadDir;
-
-    /**
-     * @param SettingsRepositoryInterface $settings
-     * @param FilesystemInterface $uploadDir
+     * {@inheritdoc}
      */
     public function __construct(SettingsRepositoryInterface $settings, FilesystemInterface $uploadDir)
     {
-        $this->settings = $settings;
-        $this->uploadDir = $uploadDir;
+        parent::__construct($settings, $uploadDir);
+
+        $this->filenamePrefix = 'favicon';
```

Comment:
Same here, if these are directly defined as class properties we don't need to redeclare the constructor

Reply:
All done, tests are green, thanks for the review!